### ISSUE TITLE
linter: react/display-name false positive for exported default classes

### ISSUE THREAD

**Issue by RioShopfront** [oxc-project/oxc#21632]
### What version of Oxlint are you using?

1.6.1

### What command did you run?

`oxlint`

### What does your `.oxlintrc.json` (or `oxlint.config.ts`) config file look like?

```ts
export default defineConfig({
    plugins  : [ "react" ],
    rules: {
        "react/display-name": "error"
    }
})
```

### What happened?

The rule produces false errors for exported, default classes which extend React.Component

The below code produces false errors. There are no current tests cases that cover this case.
```tsx
export default class Logo extends React.Component<Props> {
    public url = "/imgs/image.png?cache=true";

    public render(): React.ReactNode {
        return (
            <Link to="/" onClick={this.props.onClick} id="logo">
                <img
                    src={CDN(this.url)}
                    height="25"
                />
            </Link>
        );
    }
}
```

This code is accepted by the rule
```tsx
class Logo extends React.Component<Props> {
    public url = "/imgs/image.png?cache=true";

    public render(): React.ReactNode {
        return (
            <Link to="/" onClick={this.props.onClick} id="logo">
                <img
                    src={CDN(this.url)}
                    height="25"
                />
            </Link>
        );
    }
}

export default Logo;
```
